coordinator/src/main/scala/com/socrata/datacoordinator/resources/DatasetResource.scala

```diff
@@ -0,0 +1,219 @@
+          case Right((events, boundResetter)) =>
+            val iteratorOrError = try {
+              Right(JsonArrayIterator[JValue](events))
+            } catch { case _: JsonBadParse =>
+              Left(datasetBadRequest(BodyRequestError.NOT_JSON_ARRAY))
+            }
+            iteratorOrError match {
+              case Right(iterator) =>
+                val ProcessMutationReturns(copyNumber, version, lastModified, result) =
+                  processMutation(datasetId, iterator.map { ev => boundResetter(); mutateRate.mark(); ev }, tmp)
+                OK ~>
+                  ContentType(JsonContentType) ~>
+                  Header("X-SODA2-Truth-Last-Modified", dateTimeFormat.print(lastModified)) ~>
+                  Header("X-SODA2-Truth-Version", version.toString) ~>
+                  Header("X-SODA2-Truth-Copy-Number", copyNumber.toString) ~>
+                  Stream { w =>
+                    val bw = new BufferedOutputStream(w)
+                    bw.write('[')
+                    result.foreach(new Function1[MutationScriptCommandResult, Unit] {
+                      var didOne = false
+                      def apply(r: MutationScriptCommandResult) {
+                        if(didOne) bw.write(',')
+                        else didOne = true
+                        writeResult(bw, r, tmp)
+                      }
+                    })
+                    bw.write(']')
+                    bw.flush()
+
+                    log.debug("Non-linear index seeks: {}", tmp.stats.nonLinearIndexSeeks)
+                    log.debug("Non-linear data seeks: {}", tmp.stats.nonLinearDataSeeks)
+                  }
+              case Left(error) =>
+                error
+            }
+          case Left(response) =>
+            response
+        }
+      }
+      responseBuilder(resp)
+    }
+  }
+
+  def doDeleteDataset(req: HttpRequest): HttpResponse = {
+    deleteDataset(datasetId) match {
+      case DatasetDropper.Success =>
+        OK ~>
+          Header("X-SODA2-Truth-Last-Modified", dateTimeFormat.print(DateTime.now)) ~>
+          Header("X-SODA2-Truth-Copy-Number", 0.toString) ~>
+          Header("X-SODA2-Truth-Version", 0.toString) ~>
+          Content(JsonContentType, "[]")
+      case DatasetDropper.FailureNotFound =>
+        notFoundError(datasetId)
+      case DatasetDropper.FailureWriteLock =>
+        writeLockError(datasetId)
+    }
+  }
+
+
+
+  def doExportFile(req: HttpRequest): HttpResponse = {
+    val servReq = req.servletRequest
+    val precondition = req.precondition
+    val schemaHash = Option(servReq.getParameter("schemaHash"))
+    val ifModifiedSince = req.dateTimeHeader("If-Modified-Since")
+    val onlyColumns = Option(servReq.getParameterValues("c")).map(_.flatMap { c =>
+      norm(c).split(',').map(new UserColumnId(_)) }).map(UserColumnIdSet(_ : _*))
+    val limit = Option(servReq.getParameter("limit")).map { limStr =>
+      try {
```

Comment:
Is it possible to use Scala's [Try](http://www.scala-lang.org/files/archive/nightly/docs/library/index.html#scala.util.Try) for situations like these?